pkg/sharings/sharings.go

```diff
@@ -0,0 +1,140 @@
+package sharings
+
+import (
+	"github.com/cozy/cozy-stack/pkg/consts"
+	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/permissions"
+	"github.com/cozy/cozy-stack/pkg/utils"
+	"github.com/cozy/cozy-stack/web/jsonapi"
+)
+
+// Sharing contains all the information about a sharing
+type Sharing struct {
+	SID         string `json:"_id,omitempty"`
+	SRev        string `json:"_rev,omitempty"`
+	Type        string `json:"type,omitempty"`
+	Owner       bool   `json:"owner"`
+	Desc        string `json:"desc,omitempty"`
+	SharingID   string `json:"sharing_id,omitempty"`
+	SharingType string `json:"sharing_type"`
+
+	Permissions *permissions.Set    `json:"permissions,omitempty"`
+	SRecipients []*SharingRecipient `json:"recipients,omitempty"`
```

Comment:
For files, we have:

- ids and types in `data.relationships`
- full content in `included`

For sharing, I think we currently have:

- ids and types in `data.relationships`
- full content in `included`
- but also full content in `data.attributes.recipients`

If it is the case, we should avoid having `data.attributes.recipients`.